src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -137,6 +207,7 @@ async def _handle_unary_unary_rpc(object method_handler,
         SendMessageOperation(response_raw, _EMPTY_FLAGS),
     )
     await execute_batch(rpc_state, send_ops, loop)
+    rpc_state.status_sent = True
```

Comment:
Should we move this line before the `execute_batch`? So following the same pattern that you already used here [1]?.

This is IMO the most effective way - that does not require a lock - for not having data races where one abort/read/write operation and a finalization status operation are tried to send at the same time.

[1] https://github.com/grpc/grpc/pull/21582/files#diff-b53c077f7911f53dc8ce7656e16d35cdR86

Reply:
Make sense! Changed throughout this PR.